### ISSUE TITLE
Document supported operating systems.

### ISSUE THREAD

**Issue by electrovir** [asdf-vm/asdf#1687]
### How can we improve the documentation?

I've searched both the [documentation website](https://asdf-vm.com) and GitHub issues and I see no clear documentation as to what operating systems are supported.

I'm guessing that, based on the table [on the Getting Started page](https://asdf-vm.com/guide/getting-started.html) and relevant GitHub issues that I _have_ found, that Windows is totally not supported but WSL2 is mostly supported, but I'm still unsure of any of that because it's not explicitly stated anywhere.

**Comment by hyperupcall** [comment]
You're right, that part could be documented better. The most information is at #1687 and #450. Windows is not supported directly, and neither it is via Cygwin or MSYS2. asdf should Just Work with WSL2 - the tests mostly pass, but we don't officially support it since that would require adding the platform to CI somehow (for something to be officially supported, it must be tested in CI)

**Comment by hyperupcall** [comment]
As for other UNIX-like systems, we only support Linux. There has been requests for FreeBSD (see #1605), but that requires extra work that has not been done yet
